### DISCUSSION TITLE
Please help me with the scraper

### DISCUSSION THREAD

**Discussion by pthanhtu92** [stashapp/stash#4558] [Q&A] [answered]
Hi. I just updated to the newest version of Stash. I installed all the scraper though the Stash web interface as recommended. Now I can't use scraper. I use IAFD scraper and I showed "could not unmarshal json from script output: EOF". In the log said:  `ERRO[2024-02-14 13:54:31] [Scrape / IAFD] Traceback (most recent call last):
ERRO[2024-02-14 13:54:31] [Scrape / IAFD]   File "G:\stash\scrapers\community\IAFD\IAFD.py", line 63, in <module>
ERRO[2024-02-14 13:54:31] [Scrape / IAFD]     values: Iterable[str], f: Callable[[str], (T | None)] = lambda x: x
ERRO[2024-02-14 13:54:31] [Scrape / IAFD] TypeError: unsupported operand type(s) for |: 'TypeVar' and 'NoneType'
ERRO[2024-02-14 13:54:31] could not unmarshal json from script output: EOF
ERRO[2024-02-14 13:54:31] scrapeSceneURL: input: scrapeSceneURL could not unmarshal json from script output: EOF
2024-02-14T13:54:31.7543194-05:00 INF Response: 200 OK httpRequest={"proto":"HTTP/1.1","remoteIP":"127.0.0.1:55692","requestID":"Bee/BNSXvQRdSW-000676","requestMethod":"POST","requestPath":"/graphql","requestURL":"http://127.0.0.1:9999/graphql"} httpResponse={"bytes":130,"elapsed":316.1857,"status":200} service=stash`

**Comment by Maista6969** [comment, marked as answer]
This scraper requires as newer version of Python than you have installed: I recommend upgrading to Python 3.12 or 3.11

**Reply by pthanhtu92** [reply]
I currently have Python 3.9.5.

**Reply by pthanhtu92** [reply]
I installed python 3.12.2. I restarted and confirmed via command line that I have python 3.12.2, but the scraper still doesn't work. This time the error a little bit different. `ERRO[2024-02-14 22:12:54] [Scrape / IAFD] Traceback (most recent call last):
ERRO[2024-02-14 22:12:54] [Scrape / IAFD]   File "G:\stash\scrapers\community\IAFD\IAFD.py", line 6, in <module>
ERRO[2024-02-14 22:12:54] [Scrape / IAFD]     import requests
ERRO[2024-02-14 22:12:54] [Scrape / IAFD] ModuleNotFoundError: No module named 'requests'
ERRO[2024-02-14 22:12:54] could not unmarshal json from script output: EOF
ERRO[2024-02-14 22:12:54] scrapeSceneURL: input: scrapeSceneURL could not unmarshal json from script output: EOF
2024-02-14T22:12:54.1001279-05:00 INF Response: 200 OK httpRequest={"proto":"HTTP/1.1","remoteIP":"127.0.0.1:50721","requestID":"Bee/BqqUrO2ZLf-000271","requestMethod":"POST","requestPath":"/graphql","requestURL":"http://127.0.0.1:9999/graphql"} httpResponse={"bytes":130,"elapsed":124.5404,"status":200} service=stash`

**Reply by pthanhtu92** [reply]
I figured it out. I installed cloudscraper and lxml. Thank you for your help
